test/src/code_field/code_controller_modify_selected_lines_test.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter_code_editor/flutter_code_editor.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:highlight/languages/java.dart';
+
```

Comment:
Cover more cases:

Non-folded:
- start = -1, end = -1
- start = 0, end = 0 (start of the document)
- collapsed at start of a non-first line.
- collapsed at mid-line, at a column that is a multiple of tab length. (2 -> 4)
- collapsed at mid-line, at a column that is not a multiple of tab length. (3 -> 4)
- collapsed at the end of the document.
- the entire multi-line document selected.
- non-collapsed selection, two lines, not first, not last.

Folded:

![image](https://user-images.githubusercontent.com/44893228/209779261-ba82e525-fd23-4960-81d9-a727dc58875a.png)

```dart
int n;
void main() {
  if (true) {
    print('true');
  }
}
int m;
```